src/main/resources/openapi/EducationAndWorkPlanAPI.yml

```diff
@@ -0,0 +1,278 @@
+          description: An ISO-8601 timestamp representing when the Goal was last updated. This will be the same as the created date if it has not yet been updated.
+          example: '2023-06-19T09:39:44Z'
+      required:
+        - goalReference
+        - title
+        - reviewDate
+        - status
+        - createdBy
+        - createdAt
+        - updatedBy
+        - updatedAt
+    StepResponse:
+      title: StepResponse
+      type: object
+      description: A Step within a Goal that must be completed in order for the overall Goal to be completed.
+      properties:
+        stepReference:
+          type: string
+          description: A unique reference for the Step
+          example: d38a6c41-13d1-1d05-13c2-24619966119b
+        title:
+          type: string
+          description: A title describing the step
+          example: Book communication skills course
+        targetDate:
+          type: string
+          format: date
+          description: A date by which the prisoner should aim to complete the step.
+          example: '2023-08-19'
+        status:
+          $ref: '#/components/schemas/StepStatus'
+        sequenceNumber:
+          type: integer
+          description: The number (position) of the Step within the overall Goal.
+          example: 1
+
+    CreateGoalRequest:
+      title: Goal
+      type: object
+      description: A request to create an individual Goal that a Prisoner aims to complete as part of their Action Plan. The goal's status will be set to 'ACTIVE'.
+      properties:
+        title:
+          type: string
+          description: A title explaining the aim of the goal.
+          example: Improve communication skills
+        reviewDate:
+          type: string
+          format: date
+          description: An ISO-8601 date representing when the Goal is up for review. Defaults to 6 months.
+          example: '2023-12-19'
+        steps:
+          type: array
+          description: A List of at least one Step.
+          items:
+            $ref: '#/components/schemas/CreateStepRequest'
+        notes:
+          type: string
+          description: Some additional notes related to the Goal.
+          example: Pay close attention to Peter's behaviour.
+      required:
+        - title
+        - reviewDate
+    CreateStepRequest:
+      title: CreateStepRequest
+      type: object
+      description: A request to create a Step within a Goal. The Step is created a status of NOT_STARTED.
+      properties:
+        title:
+          type: string
+          description: A title describing the step
+          example: Book first aid course
+        targetDate:
+          type: string
+          format: date
+          description: A date by which the prisoner should aim to complete the step.
+          example: '2023-12-19'
+        sequenceNumber:
+          type: integer
+          description: The number (position) of the Step within the overall Goal.
+          example: 1
```

Comment:
Any required properties for `CreateStepRequest` ?
Also, sequence number .... I wonder if we just simply add it to the bottom of the list in the domain, and let the domain class(es) sequence it and assign it the correct sequence number?